static/js/stream_popover.js

```diff
@@ -431,6 +435,32 @@ export function register_stream_handlers() {
         e.stopPropagation();
     });
 
+    // Unstar all messages in a particular topic
+    $("body").on("click", ".sidebar-popover-unstar-all-in-topic", (e) => {
+        e.preventDefault();
+        e.stopPropagation();
+
+        const topic_name = $(".sidebar-popover-unstar-all-in-topic").attr("data-topic-name");
+        const stream_id = $(".sidebar-popover-unstar-all-in-topic").attr("data-stream-id");
+        const args = {
+            topic_name,
+            stream_id,
+        };
+
+        $(".left-sidebar-modal-holder").empty();
+        $(".left-sidebar-modal-holder").html(render_unstar_messages_modal(args));
+        $("#unstar-messages-modal").modal("show");
+        hide_topic_popover();
+    });
+
+    $("body").on("click", "#do_unstar_messages_in_topic_button", (e) => {
+        const topic_name = $("#unstar-messages-modal").attr("data-topic-name");
+        const stream_id = $("#unstar-messages-modal").attr("data-stream-id");
```

Comment:
I wonder if it'd be better to store these values in a lambda/continuation, rather than in the DOM this way -- I guess that'd be easy to od if the `unstar-messages-modal` wasn't a singleton (and we could just register the click handler inside the function that pops it open).

Reply:
Both the unstar all messages and unstar all messages in topic systems now use the `confirm_dialog.js` widget! This took a lot of code out of `stream_popover.js`, which anyways wasn't related to the stream popovers directly.